.evergreen/config.yml

```diff
@@ -1824,87 +1830,94 @@ axes:
   - id: os-ssl-legacy
     display_name: OS
     values:
-      - id: "ubuntu1404-go-1-15"
+      - id: "ubuntu1404-go-1-16"
         display_name: "Ubuntu 14.04"
         run_on: ubuntu1404-test
         variables:
-          GO_DIST: "/opt/golang/go1.15"
+          GO_DIST: "/opt/golang/go1.16"
+          # The GCC toolchain that comes with Ubuntu 14.04 is incompatible with the standard binary
+          # release of Go 1.16. As a result, Cgo builds on Ubuntu 14.04 fail with a linking error
+          # (see https://github.com/golang/go/issues/43996). On Ubuntu 14.04, we only run tests for
+          # server v2.6, which doesn't support client-side encryption. Exclude all build tags to
+          # prevent attempting to link the client-side encryption (libmongocrypt) binaries when
+          # running Go tests.
+          GO_BUILD_TAGS: ""
 
   # OSes that require >= 3.2 for SSL
   - id: os-ssl-32
     display_name: OS
     values:
-      - id: "windows-64-go-1-15"
+      - id: "windows-64-go-1-16"
         display_name: "Windows 64-bit"
         run_on:
           - windows-64-vs2017-test
         variables:
           GCC_PATH: "/cygdrive/c/ProgramData/chocolatey/lib/mingw/tools/install/mingw64/bin"
-          GO_DIST: "C:\\golang\\go1.15"
+          GO_DIST: "C:\\golang\\go1.16"
           PYTHON3_BINARY: "C:/python/Python38/python.exe"
           VENV_BIN_DIR: "Scripts"
-      - id: "ubuntu1604-64-go-1-15"
+      - id: "ubuntu1604-64-go-1-16"
         display_name: "Ubuntu 16.04"
         run_on: ubuntu1604-build
         variables:
-          GO_DIST: "/opt/golang/go1.15"
+          GO_DIST: "/opt/golang/go1.16"
           PYTHON3_BINARY: "/opt/python/3.8/bin/python3"
-      - id: "osx-go-1-15"
+      - id: "osx-go-1-16"
         display_name: "MacOS 10.14"
         run_on: macos-1014
         variables:
-          GO_DIST: "/opt/golang/go1.15"
+          GO_DIST: "/opt/golang/go1.16"
```

Comment:
I echo @benjirewis in waiting to run tasks on each changed platform before merging. macOS in particular is a special case, as it is not hosted through AWS, but instead with macstadium. We cannot spawn macOS hosts directly through evergreen. So investigating an issue with a macOS host is a little more work.

Reply:
Patch LGTM!